File: rosalind/SUBS/subs.rb
def substring_indices(string, substring, offset, indices)
  new_index = string.index(substring, offset)
  return substring_indices(string, substring, new_index + 1, indices << new_index + 1) if(new_index)
  indices
end

p substring_indices("ACGTACGTACGTACGT", "GTA", 0, [])

File: rosalind/SUBS/subs.clj
(defn substring-indices [string substring]
  (loop [offset 0
         indices []]
    (let [new-index (.indexOf string substring offset)]
      (if (not (neg? new-index))
        (recur (inc new-index) (conj indices (inc new-index)))
        indices))))

(println (substring-indices (str (read)) (str (read))))
